### ISSUE TITLE
Breaks Chai all.keys

### ISSUE THREAD

**Issue by ThomWright** [chaijs/chai-things#24]
Chai's built-in `all.keys` assertions seem to break when I use `chai-things` - [Chai keys](http://chaijs.com/api/bdd/#keys-section)

```js
expect({a: "string", b: "string", c: "string"}).to.have.all.keys(["a", "b"])
```

`AssertionError: expected { Object (a, b, ...) } to have a property 'length'`

**Comment by keithamus** [comment]
Hey @ThomWright thanks for the issue.

This does seem to be a problem. I'd welcome a PR to fix this if you have the time :smile:

**Comment by JohnBaileyN** [comment]
This is sad :(

**Comment by ThomWright** [comment]
PR sent. The Travis build failed, but it seems to be because it hasn't been configured and is trying to build a Ruby project.

**Comment by jamespedid** [comment]
This might be a breaking change, but perhaps instead of using the word all, you use the word each.

expect(collection).to.each.have.property("foo", "bar")

instead of 

expect(collection).to.each.have.property("foo", "bar")

This makes more sense grammatically as well.

**Comment by ThomWright** [comment]
:+1: That would probably be a better solution.

**Comment by lo1tuma** [comment]
I’ve exactly the same problem. The `key`/`keys` assertion isn’t usable at all if `chai-things` is used.

**Comment by wassname** [comment]
I agree that each is more intuitive than all, seems like a decent idea. You could even make it work for object values.

A workaround for this is omit writing "all", since chai defaults to using it anyway. From the chai docs:

> Note, either any or all should be used in the assertion. If neither are used, the assertion is defaulted to all. [[1](http://chaijs.com/api/bdd/#method_keys)]

**Comment by drzaal** [comment]
I tried to fix this. I might have gotten a bit overaggressive with it though.

Side effects include allowing testing over enumerable object properties
and the inclusion of synonym 'each'

https://github.com/chaijs/chai-things/pull/35

**Comment by wassname** [comment]
Nice!

**Comment by drzaal** [comment]
Get unexcited. It's clear there are some implementation decisions in this module which are at odds with the chai implementations.

The 'contains' flag interaction is mandatory for 'any' and has no effect on 'all'

however in default chai the behavior is
'contains' has no effect on 'any', but determines whether 'all' is inclusive or comprehensive.

There is also a difference between the included chai package and most up to date package of chai.

**Comment by drzaal** [comment]
Okay. It's ready again. Whew.
#35

**Comment by piotr-s-brainhub** [comment]
+1

**Comment by poupryc** [comment]
I would like to ask for news of the issue ^^

**Comment by hypeofpipe** [comment]
Seems it doesn't work at all.

**Comment by master-of-null** [comment]
It's pretty bad that this is 4+ years old. Just used `chai-things` and now have to revert. Didn't notice it wasn't maintained

**Comment by JonnoFTW** [comment]
If they aren't going to maintain this plugin it needs to be marked as archived and alternatives given for its use cases.

**Comment by JamieMcNaught** [comment]
Sadly the broken example of this still exists in the Chai documentation:
https://www.chaijs.com/api/bdd/#method_all
